driver.c

```diff
@@ -1791,16 +1791,15 @@ static bool driver_setup (settings_t *settings)
         z_step_pio = pio1;
     step_pulse_program_init(z_step_pio, z_step_sm, pio_offset, Z_STEP_PIN, 1);
 
-#if N_ABC_MOTORS > 1
+#if N_ABC_MOTORS > 0
 
 #if WIFI_ENABLE && N_ABC_MOTORS > 2
 #error "Max number of motors with WIFI_ENABLE is 5"
 #endif
 
-    if(++step_sm > 3) {
-        step_sm = 0;
-        pio_offset = pio_add_program(pio0, &step_pulse_program);
-    }
```

Comment:
IMO these lines should be left "as-is" since pio_add_program(pio0, ...) may already have been called at line 1789 if compiling for Pico W. If not sm 0 will be assigned to two motors.

Reply:
Thanks for catching this. On closer inspection, yes, z logic is handled very specially. For clarity it may be better to store a pio # for each axis and have more generic logic for adding each stepper in sequence to avoid errors, but I will do a minimal fix here pending further instructions.